src/plugins/links/public/embeddable/links_embeddable.tsx

```diff
@@ -6,152 +6,166 @@
+
+      // by-reference embeddable
+      if (!('attributes' in serializedState) || serializedState.attributes === undefined) {
+        return serializedState;
+      }
+
+      const { attributes: attributesWithInjectedIds } = injectReferences({
+        attributes: serializedState.attributes as unknown as LinksAttributes,
+        references: state.references ?? [],
+      });
+
+      return {
+        ...serializedState,
+        attributes: attributesWithInjectedIds,
+      };
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+      const { linksApi, linksComparators, serializeLinks } = await initializeLinks(
+        state,
+        uuid,
+        parentApi
+      );
+
+      const serializeState = (): SerializedPanelState<LinksSerializedState> => {
+        const newState = {
+          ...state,
+          ...serializeTitles(),
+          ...serializeLinks(),
+        };
+        // by-reference embeddable
+        if (!('attributes' in newState) || newState.attributes === undefined) {
+          // No references to extract for by-reference embeddable since all references are stored with by-reference saved object
+          return { rawState: newState, references: [] };
+        }
+
+        // by-value embeddable
+        const { attributes, references } = extractReferences({
+          attributes: newState.attributes,
+        });
+
+        return {
+          rawState: { ...state, attributes },
+          references,
+        };
+      };
+
+      const isByReference = state.savedObjectId !== undefined;
+
+      const api = buildApi(
+        {
+          ...titlesApi,
+          ...linksApi,
+          ...intializeLibraryTransforms(linksApi.attributes$.value, serializeState, isByReference),
+          getTypeDisplayName: () => APP_NAME,
+          serializeState,
+        },
+        {
+          ...linksComparators,
+          ...titleComparators,
+          enhancements: getUnchangingComparator(),
+          disabledActions: getUnchangingComparator(),
+        }
+      );
+
+      const Component = () => {
+        const [resolvedLinks, attributes] = useBatchedPublishingSubjects(
+          api.resolvedLinks$,
+          api.attributes$
+        );
+
+        const layout = attributes?.layout ?? LINKS_VERTICAL_LAYOUT;
+
+        const linkItems: { [id: string]: { id: string; content: JSX.Element } } = useMemo(() => {
+          return resolvedLinks.reduce((prev, currentLink) => {
+            return {
+              ...prev,
+              [currentLink.id]: {
+                id: currentLink.id,
+                content:
+                  currentLink.type === DASHBOARD_LINK_TYPE ? (
+                    <DashboardLinkComponent
+                      key={currentLink.id}
+                      link={currentLink}
+                      layout={layout}
+                      api={api}
```

Comment:
Maybe it's cleaner to pass in the `parentApi` directly here, since it seems like that is the only part we are using?

Reply:
`DashboardLinkComponent` also uses the full `api` in the `getDashboardLocatorParamsFromEmbeddable` function which needs the `api.timeRange$.value`.

Reply:
In this case, it makes more sense to default to the parent API's `timeRange` anyway, right? Since the **link** isn't providing a time range, it's the parent (dashboard) that is. Which is what happens in `getDashboardLocatorParamsFromEmbeddable`:

```typescript
  const timeRange = api.timeRange$?.value ?? api.parentApi?.timeRange$?.value;
  if (timeRange && options.useCurrentDateRange) {
    params.timeRange = timeRange;
  }
```

In some of my discussions with Devon, it seems like we should be trying to limit the places we are passing in the full API... But again, just a nit :) It might be a bit ugly in this case, because we would have to pass in the API in `getDashboardLocatorParamsFromEmbeddable` as something like:

```typescript
getDashboardLocatorParamsFromEmbeddable({ parentApi })
```

which isn't super nice either. For now, I'd be fine with leaving it as-is if you feel strongly :+1: